pkg/v2/tkr/webhook/infra-machine/fieldsetter/fieldsetter.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package fieldsetter defines methods to set the fields of <Infra>Machine resource  .
+package fieldsetter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	osImageRefAnnotationKey = "run.tanzu.vmware.com/os-image-ref"
+)
+
+// FieldSetter mutates <infra>Machine
+type FieldSetter struct {
+	decoder      *admission.Decoder
+	Log          logr.Logger
+	FieldPathMap map[string]string
+}
+
+// Handle will handle the infra admission request
+func (fs *FieldSetter) Handle(ctx context.Context, req admission.Request) admission.Response { // nolint:gocritic //req parameter is not pointer type as per standard signature of webhook callback handler
+	fs.Log.Info("Received the infra machine admission request")
+	infraMachine := &unstructured.Unstructured{}
+	err := fs.decoder.Decode(req, infraMachine)
+	if err != nil {
+		fs.Log.Error(err, "Failed to decode infra machine admission request")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	annotationValues, err := getOSImageRefAnnotation(infraMachine)
+	if err != nil {
+		fs.Log.Error(err, "Failed to get OSImageref annotation")
+		return admission.ValidationResponse(true, err.Error())
```

Comment:
We should *not* allow the request if we got an error.

Reply:
same as explained for comment at L42